truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java

```diff
@@ -119,14 +127,31 @@ public NodeCost getCost() {
     }
 
     /**
-     * Clears any previously assigned guest language source code from this node.
+     * @deprecated if your node provides source section, override {@link #getSourceSection()} to
+     *             return it - this method will be removed
+     * 
+     *             Clears any previously assigned guest language source code from this node.
      */
+    @Deprecated
     public void clearSourceSection() {
         this.sourceSection = null;
     }
 
     /**
-     * Retrieves the segment of guest language source code that is represented by this Node.
+     * Retrieves the segment of guest language source code that is represented by this Node. The
+     * default implementation of this method returns <code>null</code>. If your node represents a
+     * segment of the source code, override this method and return a <code>final</code> or
+     * {@link CompilationFinal} field in your node to the caller.
+     *
+     * To define node with <em>fixed</em> {@link SourceSection} that doesn't change after node
+     * construction use:
+     *
+     * {@codesnippet source.section.fixed}
+     *
+     * To create a node which can associate and change the {@link SourceSection} later at any point
+     * of time use:
+     *
+     * {@codesnippet source.section.mutable}
      *
      * @return the source code represented by this Node
      */
```

Comment:
Does this comment need a transitory note? Mostly, because the implementation does not yet return null?